csunplugged/utils/TranslatableModelLoader.py

```diff
@@ -0,0 +1,166 @@
+                        raise MissingRequiredFieldError(
+                            translations_filename,
+                            [required_field],
+                            "model ({})".format(model_slug)
+                        )
+                for field, value in model_fields.items():
+                    if not isinstance(value, str):
+                        raise InvalidConfigValueError(
+                            translations_filename,
+                            "{}->{}".format(model_slug, field),
+                            "String"
+                        )
+                    if field_map:
+                        field = field_map.get(field, field)
+                    values_dict[field] = value
+                translations.setdefault(model_slug, dict())[language] = values_dict
+        return translations
+
+    def get_markdown_translations(self, filename, required=True, **kwargs):
+        """Get dictionary of translations of the requested markdown file.
+
+        Args:
+            filename: (str) path to yaml file from the working directory of the loader
+            required: (bool) raise an exception if the requested file is not present
+                in the /en directory tree
+            kwargs: (dict) kwargs passed through to convert_md_file
+
+        Returns:
+            dict mapping language codes to VertoResult objects
+
+        Raises:
+            CouldNotFindMarkdownFileError if the requested file could not be found
+                in the /en directory tree
+        """
+        content_translations = {}
+        for language in get_available_languages():
+            try:
+                content_translations[language] = self.convert_md_file(
+                    self.get_localised_file(
+                        language,
+                        filename,
+                    ),
+                    self.structure_file_path,
+                    **kwargs
+                )
+            except CouldNotFindMarkdownFileError:
+                if required and language == get_default_language():
+                    raise
+        return content_translations
+
+    @staticmethod
+    def populate_translations(model, model_translations_dict):
+        """Populate the translation fields of the given model.
+
+        Args:
+            model: TranslatableModel instance
+            model_translations_dict: dictionary of form
+                {
+                    <language_code>: {
+                        <field_name>:<value>,
+                        <field_name>:<value>,...
+                    },...
+                }
+        """
+        for language, values_dict in model_translations_dict.items():
+            with translation.override(language):
+                for field, value in values_dict.items():
+                    setattr(model, field, value)
+
+    @staticmethod
+    def mark_translation_availability(model, required_fields=[]):
```

Comment:
why does `required_fields` have a default value/is there a situation where this method can be called without any fields needing translation?

Reply:
As discussed offline, an empty list here will mean the model is available in all languages, regardless of whether any of its fields are translated. While I can't currently think of a good reason to use this, preventing it seems needlessly restrictive.

I've updated to docstring to make this clearer.